frontend/src/pages/Blogs/EditBlog.jsx

```diff
@@ -0,0 +1,278 @@
+import parse from 'html-react-parser';
+
+import EditBlogModal from '../../components/EditBlog/EditBlogModal';
+import { EDITOR_API_KEY, EDITOR_INIT } from '../../utils/editorConstants';
+
+// Styles
+const Container = Styled.div`
+  margin: 0 10%;
+`;
+
+const BlogContainer = Styled.div`
+  margin: 5% 0;
+`;
+
+const HeaderContainer = Styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  margin: 10px 0;
+`;
+
+const Heading = Styled.p`
+  font-size: ${props => props.theme.fontSizes.h1};
+  font-weight: bold;
+  margin: 0;
+`;
+
+const ActionButtonContainer = Styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const InputContainer = Styled.div`
+  margin: 10px 0;
+`;
+
+const TitleInput = Styled(Input)`
+  width: 100%;
+  font-size: ${props => props.theme.fontSizes.h3};
+`;
+
+const TagsInput = Styled(Input)`
+  width: 100%;
+  font-size: ${props => props.theme.fontSizes.p};
+`;
+
+const ActionButton = Styled.div`
+  display: flex;
+  background-color: ${props => props.theme.colors.lightPurple};
+  font-size: ${props => props.theme.fontSizes.p};
+  color: ${props => props.theme.colors.white};
+  padding: 15px 40px;
+  cursor: pointer;
+  max-width: 250px;
+  max-height: 75px;
+  align-items: center;
+  justify-content: center;
+  border-radius: 8px;
+  font-weight: bold;
+  margin: 10px 0;
+
+  &:hover {
+    opacity: 70%;
+  }
+`;
+
+const UploadButtonText = Styled.p`
+  color: ${props => props.theme.colors.white};
+  font-size ${props => props.theme.fontSizes.p};
+  font-weight: bold;
+`;
+
+const PublishButtonContainer = Styled.div`
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const PreviewBlogTitle = Styled.p`
+  font-size: ${props => props.theme.fontSizes.h2};
+  font-weight: bold;
+  margin-bottom: 16px;
+`;
+
+class EditBlog extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      action: 'save',
+      blogTitle: '',
+      coverPhoto: '',
+      editorContent: '',
+      modalOpen: false,
+      isEditing: true
+    };
+    this.editor = React.createRef();
+  }
+
+  /* MODAL CALLBACKS */
+
+  onPressSave = () => {
+    this.setState({ modalOpen: false });
+  };
+
+  onPressCancel = () => {
+    this.setState({ modalOpen: false });
+  };
+
+  /* CLASS FUNCTIONS */
+  onEditorChange = content => {
+    this.setState({ editorContent: content });
+  };
+
+  onPressUploadImage = () => {
+    this.setState({
+      modalOpen: true,
+      action: 'save'
+    });
+  };
+
+  onTitleChange = title => {
+    this.setState({ blogTitle: title.target.value });
+  };
+
+  onPressPreview = () => {
+    this.setState({isEditing: false});
+  };
+
+  onPublishBlog = () => {
+    // TODO: logic for publishing a blog
+  };
+
+  onPressKeepEditing = () => {
+    this.setState({isEditing: true});
+  }
+
+  getActionFunction() {
+    switch (this.state.action) {
+      case 'save':
+        return this.onPressSave;
+        break;
```

Comment:
nit: these breaks are technically unreachable and create a warning when you first launch the frontend. Can you remove them?

Reply:
👍